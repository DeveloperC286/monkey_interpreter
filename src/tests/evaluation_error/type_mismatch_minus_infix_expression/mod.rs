use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case("TRUE - TRUE", "type_mismatch_minus_infix_expression_case_1"),
    case("10 - TRUE", "type_mismatch_minus_infix_expression_case_2"),
    case(
        "true - \"this is a string\"",
        "type_mismatch_minus_infix_expression_case_3"
    ),
    case("\"123\" - 10", "type_mismatch_minus_infix_expression_case_4"),
    case("7 - \"456\"", "type_mismatch_minus_infix_expression_case_5")
)]
fn type_mismatch_minus_infix_expression_cases(code: &str, snapshot_name: &str) {}

#[apply(type_mismatch_minus_infix_expression_cases)]
fn test_type_mismatch_minus_infix_expression_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(type_mismatch_minus_infix_expression_cases)]
fn test_type_mismatch_minus_infix_expression_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(type_mismatch_minus_infix_expression_cases)]
fn test_type_mismatch_minus_infix_expression_evaluation_error(code: &str, snapshot_name: &str) {
    assert_evaluation_error!(code, snapshot_name);
}
