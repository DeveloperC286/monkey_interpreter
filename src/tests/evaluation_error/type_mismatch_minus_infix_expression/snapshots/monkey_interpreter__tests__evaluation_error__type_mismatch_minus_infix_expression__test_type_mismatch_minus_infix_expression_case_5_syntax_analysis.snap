---
source: src/tests/evaluation_error/type_mismatch_minus_infix_expression/mod.rs
expression: abstract_syntax_tree
---
[
    Expression {
        expression: MinusInfix {
            left_hand: String {
                literal: "456",
            },
            right_hand: Integer {
                literal: 7,
            },
        },
    },
]
